packages/message-relayer/src/service.ts

```diff
@@ -262,7 +262,12 @@ export class MessageRelayerService extends BaseService<MessageRelayerOptions> {
           /* parse this to make sure that the mesaage was actually relayed */
           // clear out buffer only if the messages are relayed to L1 successfully
           if (
-            await this._wasMessageRelayed(this.state.messageBuffer[0].message)
+            await this._wereMessagesRelayed(
+              this.state.messageBuffer.reduce((acc, cur) => {
+                acc.push(cur.message)
+                return acc
+              }, [])
+            )
```

Comment:
this should reset the buffer everytime, so don't think there's a possibility, but to share the thought- if somehow a message sticks in the buffer, could make the batch revert